src/SqlPersistence/Config/ConnectionManager/ConnectionManager.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System.Collections.Generic;
+using System.Data.Common;
+using System.Threading.Tasks;
+
+abstract class ConnectionManager
+{
+    public abstract Task<DbConnection> OpenNonContextualConnection();
+    public abstract DbConnection BuildNonContextual();
+    public abstract Task<DbConnection> OpenConnection(IReadOnlyDictionary<string, string> messageHeaders);
+
+    protected async Task<DbConnection> OpenConnection(DbConnection connection)
```

Comment:
I think if we make this an extension method than we can remove `OpenNonContextualConnection` and only focus in this class on creating connections while the caller would open it via that extension method?